File: HealthcareSystem.Application/Requests/DoctorRequests.cs
namespace HealthcareSystem.Application.Requests;

public static class DoctorRequests {
    public record GetDoctorsRequest(
        int? PageIndex,
        int? PageSize,
        string? SortField,
        string? SortOrder,
        string? SearchName,
        string? SearchSpecialization
    );

    public record CreateDoctorRequest(
        string Name,
        string Description,
        string? ImageUrl,
        int ExperienceAge,
        decimal FeeInDollars,
        string Specialization,
        string PhoneNumber
    );

    public record UpdateDoctorRequest(
        string? Name,
        string? Description,
        string? ImageUrl,
        int? ExperienceAge,
        decimal? FeeInDollars,
        string? Specialization,
        string? PhoneNumber
    );
}

File: HealthcareSystem.Client/src/app/data/requests/doctorRequests.ts
export interface GetDoctorsRequest {
  pageIndex?: number;
  pageSize?: number;
  sortField?: string;
  sortOrder?: string;
  searchName?: string;
  searchSpecialization?: string;
}

export interface CreateDoctorRequest {
  name: string;
  description: string;
  imageUrl?: string;
  experienceAge: number;
  feeInDollars: number;
  specialization: string;
  phoneNumber: string;
}

export interface UpdateDoctorRequest {
  name?: string;
  description?: string;
  imageUrl?: string;
  experienceAge?: number;
  feeInDollars?: number;
  specialization?: string;
  phoneNumber?: string;
}
